### PR TITLE
feat: add new endpoints to add organizations to an existing user

### PR DESCRIPTION
**Pull request by tmy1313**
OCD-4721

### CODE REVIEW

File: chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/UserManagementController.java [hunk 1]
```diff
@@ -92,6 +96,24 @@ public UserManagementController(UserManager userManager, InvitationManager invit
         this.authorizationLengthInDays = authorizationLengthInDays;
     }
 
+    @Operation(summary = "Update the currently logged in user with an additional organization.",
+            description = "Gives the user permission on the object in the invitation (usually an additional ACB or Developer)."
```
Comment (kekey1) [inline chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/UserManagementController.java]: I think this documentation is copied from the CHPL-login workflow methods. The line about which endpoints to call in which order is not right for the Cognito methods. Can you check the other Cognito auth methods as well? They are not currently in the diff but it looks like they might have the same documentation.

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/user/cognito/CognitoApiWrapper.java [hunk 2]
```diff
@@ -181,9 +186,28 @@ public User getUserInfo(UUID cognitoId) throws UserRetrievalException {
             }
         }
         return null;
+    }
+
+    public User getUserByPassingCognitoUserAttributeCache(String accessToken) throws UserRetrievalException {
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/user/cognito/CognitoApiWrapper.java]: When I first read this method name I thought it was two words "by passing" but I think it's meant as one word "bypassing" - can you change the camelcase to make this clear? getUserBypassingCognitoUserAttributeCache, or maybe even simpler getUserNoCache?

**Review by kekey1 — DISMISSED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/user/cognito/CognitoApiWrapper.java [hunk 3]
```diff
@@ -171,39 +171,37 @@ public AuthenticationResultType respondToNewPasswordRequiredChallenge(CognitoNew
...
+        AdminGetUserResponse response = cognitoClient.adminGetUser(request);
+        if (response == null || response.sdkHttpResponse() == null || !response.sdkHttpResponse().isSuccessful()) {
+            return null;
         }
-        return null;
+        return createUserFromGetUserResponse(response);
     }
 
+
+<<<<<<< Updated upstream
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/user/cognito/CognitoApiWrapper.java]: Merge issue

**Review by kekey1 — CHANGES_REQUESTED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/user/cognito/CognitoApiWrapper.java [hunk 4]
```diff
@@ -371,6 +367,31 @@ public void updateUser(User user) throws UserRetrievalException {
         cognitoClient.adminUpdateUserAttributes(request);
     }
 
+    @CacheEvict(value = CacheNames.COGNITO_USERS, key = "#user.cognitoId")
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/user/cognito/CognitoApiWrapper.java]: It looks like we cache users both with UUID as a key and email as a key (see above method `getUserInfo(String email)`). But here we are evicting only the UUID key. I'm not 100% sure I'm correct but I think so... can you check on this?

**Review by kekey1 — CHANGES_REQUESTED** [review]

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

**Review by kekey1 — APPROVED** [review]

**Review by andlar — APPROVED** [review]

**Pull request merged**